guides/middleware.md

```diff
@@ -133,6 +133,39 @@ app.use('/message/*', async (c, next) => {
 app.get('/message/hello', (c) => c.text('Hello Middleware!'))
 ```
 
+However, embedding middleware directly within `app.use()` can limit its reusability. Therefore, we can separate our
+middleware into different files.
+
+To ensure we don't lose type definitions for `context` and `next`, when separating middleware, we can use 
+[`createMiddleware()`](/helpers/factory#createmiddleware) from Hono's factory.
+
+```ts
+import { createMiddleware } from 'hono/factory'
+
+const logger = createMiddleware(async (c, next) => {
+  console.log(`[${c.req.method}] ${c.req.url}`)
+  await next()
+})
+
+```
+Additionally, middleware can be designed to modify responses if necessary:
```

Comment:
I think it's better to separate this description from here and make it a sub-section of "Custom Middleware," which is put in.  The title will be like "Modify the Response after the `next()`".

Reply:
Sounds good, I will update that soon

Reply:
Fixed: 3abe563

Reply:
Cool!